### PR TITLE
fix support for ipywidgets 8 and remove max pin

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request attempts to support ipywidgets 8 change in `.widgets` becoming a private `._active_widgets` while still supporting the old syntax.

xref https://github.com/jupyter-widgets/ipywidgets/pull/3122

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/app.py [hunk 1]
```diff
@@ -1179,7 +1179,7 @@ def resize(stack_items):
         # resize tray items
         for tray_item in self.state.tray_items:
             # access the actual plugin object (there is no store for plugins)
-            tray_obj = self.widgets.get(tray_item['widget'].split('IPY_MODEL_')[1])
+            tray_obj = widget_serialization['from_json'](tray_item['widget'], None)
```
Comment (pllim) [inline jdaviz/app.py:1182]: Can't we create our down `widgets` property here to return either the old or new API based on ipywidgets version?

Comment (pllim) [reply jdaviz/app.py:1182]: Or maybe a new method name altogether or something?

Comment (kecnry) [reply jdaviz/app.py:1182]: I would have expected my previous commit to work if that would work.  But please feel free to try (either a separate PR or push here) as I agree that is cleaner than this!

Comment (pllim) [reply jdaviz/app.py:1182]: Mario said your solution is very nice though, so maybe you can ignore my comments.

**Comment by kecnry** [comment]
Let's squash and merge when this is approved (to get rid of the failed attempt from the commit history, but its nice to keep here in the PR so I'll avoid squashing in this branch).

**Review by pllim — APPROVED** [review]
I am approving by proxy. Maybe @havok2063 want to have a look too, just in case?

"... that’s a nice solution! It will not impact performance." -- Mario B. ⭐⭐⭐⭐⭐

**Review by havok2063 — APPROVED** [review]
Looks good to me.

**Pull request merged**